tests/pos/sams.scala

```diff
@@ -0,0 +1,75 @@
+object test {
+
+  trait X { def foo(x: Int): Int; def bar = foo(2) }
+
+  val x: X = (x: Int) => 2  // should be a closure
+
+  trait T {
+    var f = 2
+    def foo(x: Int): Int
+  }
+
+  val t: T = (x: Int) => 2   // needs to be an anonymous class because of defined field
+
+  trait U extends T
+
+  val u: U = (x: Int) => 2   // needs to be an anonymous class because of inherited field
+
+  trait V extends Exception { def foo(x: Int): Int }
+
+  val v: V = (x: Int) => 2   // needs to be an anonymous class because the trait extends a non-object class
```

Comment:
@smarter, yes, this is something else entirely, and this is what is tested here.
Quoting Martin:
>"non-object class": class other than java.lang.Object. 

In this example, `trait V` extends `class Exception`. So it could not be implemented by a JVM SAM.